pymaster/field.py

```diff
@@ -658,3 +681,176 @@ def get_ell_sampling(self):
+
+        # Can only have field or do clustering
+        if field is not None and do_clustering:
+            raise ValueError("If positions_rand and weights_rand are "
+                             "provided, field must be None.")
+
+        # Sanity checks on positions and weights
+        positions = np.array(positions, dtype=np.float64)
+        weights = np.array(weights, dtype=np.float64)
+        pw_list = [(positions, weights)]
+        if do_clustering:
+            positions_rand = np.array(positions_rand, dtype=np.float64)
+            weights_rand = np.array(weights_rand, dtype=np.float64)
+            pw_list += [(positions_rand, weights_rand)]
+        for p, w in pw_list:
+            if np.shape(p) != (2, len(w)):
+                raise ValueError("Positions must be 2D array of shape"
+                                 " (2, len(weights)).")
+            if lonlat:
+                p[0] = np.radians(90. - p[1])
+                p[1] = np.radians(p[0])
+            if not (np.logical_and(p[0] >= 0., p[0] <= np.pi)).all():
+                if lonlat:
+                    raise ValueError(
+                        "Second dimension of positions must be latitude in "
+                        "degrees, between -90 and 90."
+                    )
+                else:
+                    raise ValueError(
+                        "First dimension of positions must be colatitude in "
+                        "radians, between 0 and pi."
+                    )
+
+        # Compute mask shot noise
+        self._Nw = np.sum(weights**2.)/(4.*np.pi)
+
+        # 1. Compute mask alms and beam
+        # Sanity checks
+        if lmax_mask is None:
+            lmax_mask = 4*lmax
+        self.ainfo_mask = ut.NmtAlmInfo(lmax_mask)
+        # Mask alms
+        lmax_alm = lmax if do_clustering else lmax_mask
+        self.alm_mask = ut._catalog2alm_ducc0(weights, positions,
+                                              spin=0, lmax=lmax_alm)[0]
+        # Beam
+        if isinstance(beam, (list, tuple, np.ndarray)):
+            if len(beam) <= lmax:
+                raise ValueError("Input beam must have at least %d elements "
+                                 "given the input maximum "
+                                 "multipole" % (lmax+1))
+            beam_use = beam
+        else:
+            if beam is None:
+                beam_use = np.ones(lmax+1)
+            else:
+                raise ValueError("Input beam can only be an array or None\n")
+        self.beam = beam_use
+
+        # 2. Compute field alms
+        # If only positions and weights, just return here
+        if field is None:
+            if spin is None:
+                raise ValueError("If field, positions_rand, and weights_rand "
+                                 "are None, spin needs to be provided.")
+            if do_clustering and spin is None:
+                self.spin = 0
+            self.spin = spin
```

Comment:
OK, so here you will undo what you did in line 819. I think you need to wrap this in an `else`

Reply:
Right, see previous comment.